src/Compilers/CSharp/Test/Semantic/Semantics/SemanticErrorTests.cs

```diff
@@ -1537,16 +1537,17 @@ class C
 {
     void M1()
     {
-        object obj = true ? 0 : M2();
+        var obj = true ? 0 : M2();
```

Comment:
Why was this changed to 'var'?

Reply:
Because this test would not test what it was intended to test if there is a target type of `object`. The test would produce no diagnostics under the new rules, but the point of the test is to demonstrate this diagnostic is produced.

---
In reply to: [441790482](https://github.com/dotnet/roslyn/pull/44505#discussion_r441790482) [](ancestors = 441790482)